lib/util/query_params.dart

```diff
@@ -1,20 +1,17 @@
 import 'dart:html';
 
-class QueryParams {
+/// A utility for accessing and setting query parameters.
+abstract class QueryParams {
```

Comment:
drive-by-comment.

Maybe better to make this a final, singleton, top-level field that instantiated on first use?

It's global state either way, but it feels better.

Reply:
Thanks for the suggestion, that definitely makes sense :). I wasn't sure the most appropriate way to implement this change though, so if you wouldn't mind taking another look, that would be great.

Reply:
Love it!